lib/plaid.ex

```diff
@@ -88,33 +88,39 @@ defmodule Plaid do
   @spec make_request_with_cred(atom, String.t(), map, map, map, Keyword.t()) ::
           {:ok, HTTPoison.Response.t()} | {:error, HTTPoison.Error.t()}
   def make_request_with_cred(method, endpoint, config, body \\ %{}, headers \\ %{}, options \\ []) do
-    start_time = System.monotonic_time()
-
-    :telemetry.execute(@events_prefix ++ [:start], %{system_time: start_time}, %{
+    common_metadata = %{
       method: method,
       path: endpoint,
       u: :native
-    })
+    }
 
-    try do
-      request_endpoint = "#{get_root_uri(config)}#{endpoint}"
-      cred = Map.delete(config, :root_uri)
-      request_body = Map.merge(body, cred) |> Poison.encode!()
-      request_headers = get_request_headers() |> Map.merge(headers) |> Map.to_list()
-      options = httpoison_request_options() ++ options
+    with_metrics(
+      fn ->
+        request_endpoint = "#{get_root_uri(config)}#{endpoint}"
```

Comment:
What are your thoughts on including the preparation of the request arguments in the instrumentation? I would expect just `request/6` to be instrumented.

Reply:
Sounds good to keep as-is given the current limitations on testing. No obligation, but I'd be interested in seeing any examples you might have of Mox and how you handle this issue else.where

Reply:
Thanks, I'd like to explore implementing this. Would you be open to providing a review and some feedback of my implementation?

Also, please let me know when you'd like to turn this draft into a PR and I'll get it merged.